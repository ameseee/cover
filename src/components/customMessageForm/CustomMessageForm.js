import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { sendCustom } from '../../utils/postToServer';

class CustomMessageForm extends Component {
  constructor() {
    super();
    this.state = {
      message: '',
      sent: false
    };
  }

  handleChange = (event) => {
    this.setState({
      message: event.target.value
    });
  }

  handleSend = () => {
    sendCustom(this.state.message, '5756441355');
    this.setState({
      sent: true
    });
  }

  render() {
    const sentNotification = this.state.sent
      ? <div className="sent-icon-displayed sent-icon"></div>
      : <div className="sent-icon-hidden sent-icon"></div>;

    return (
      <div className="custom-message-form">
        <textarea
          className="custom-message-input"
          onChange={(event) => this.handleChange(event)}
          value={this.state.message}
        />
        <button
          className="close-form-btn"
          onClick={() => this.props.toggleCustomForm()}
        >
        </button>
        <article className="send-sent">
          <button
            className="send-custom-btn"
            onClick={() => this.handleSend()}
          >
            Send
          </button>
          {sentNotification}
        </article>
      </div>
    );
  }
}

CustomMessageForm.propTypes = {
  toggleCustomForm: PropTypes.func,
};

export default CustomMessageForm;
