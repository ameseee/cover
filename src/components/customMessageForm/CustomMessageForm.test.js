import React from 'react';
import CustomMessageForm from './CustomMessageForm';
import { shallow } from 'enzyme';
import testSetup from '../../../__mock__/testSetup';

describe('<CustomMessageForm />', () => {
  let wrapper;
  let mockFn = jest.fn();

  beforeEach( () => {
    wrapper = shallow(<CustomMessageForm toggleCustomForm={mockFn}/>);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have a false default state of message', () => {
    expect(wrapper.state().message).toEqual('');
  });

  it('should have a false default state of sent', () => {
    expect(wrapper.state().sent).toEqual(false);
  });

  it('should render a text area', () => {
    const text = wrapper.find('textarea');

    expect(text.length).toEqual(1);
  });

  it('should change state on input', () => {
    const message = wrapper.find('.custom-message-input');

    message.simulate('change', { target: { value: 'hello friend' } });
    expect(wrapper.state().message).toEqual('hello friend');
  });

  it('should render a button to send text', () => {
    const sendBtn = wrapper.find('.send-custom-btn');

    expect(sendBtn.length).toEqual(1);
  });

  it('should run a function to send a text', () => {
    const sendBtn = wrapper.find('.send-custom-btn');

    wrapper.instance().handleSend = mockFn;
    sendBtn.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should render a button to close form', () => {
    const closeBtn = wrapper.find('.close-form-btn');

    expect(closeBtn.length).toEqual(1);
  });

  it('should run a function to close button form', () => {
    const closeBtn = wrapper.find('.close-form-btn');

    wrapper.instance().toggleCustomForm = mockFn;
    closeBtn.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(2);
  });

  it('should render a div with a sent icon', () => {
    const sendIcon = wrapper.find('.sent-icon');

    expect(sendIcon.length).toEqual(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
