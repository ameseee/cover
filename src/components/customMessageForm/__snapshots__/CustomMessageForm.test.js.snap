// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomMessageForm /> should match snapshot 1`] = `undefined`;
